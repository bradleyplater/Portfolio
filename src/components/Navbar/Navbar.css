.navbar {
  width: 100%;
  height: 5%;
  overflow: hidden;
  background-color: #333;
}

.navbar-container {
  padding: 0;
  list-style-type: none;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-content: space-around;
  flex-wrap: wrap;
  margin: 0;
}

.navbar-list-item {
  padding: 0 2%;
  width: 20%;
}

.navbar-link {
  width: 100%;
  float: left;
  color: #f2f2f2;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 100%;
  font-family: "Architects Daughter", cursive;
}

.navbar-link:hover {
  background-color: #ddd;
  color: black;
}

.selected {
  background-color: #ddd;
  color: black;
}
