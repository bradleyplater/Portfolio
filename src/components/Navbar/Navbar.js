import React from "react";
import "./Navbar.css";
import { NavLink } from "react-router-dom";

export default function Navbar() {
  const navItems = ["Home", "Skills", "Me", "Contact"];

  return (
    <>
      <nav className="navbar">
        <ul className="navbar-container">
          {navItems.map((navItem) => (
            <li className="navbar-list-item">
              <NavLink
                to={"/" + navItem}
                data-testid={"link." + navItem}
                className="navbar-link"
                activeClassName="navbar-link selected"
              >
                {navItem}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </>
  );
}
