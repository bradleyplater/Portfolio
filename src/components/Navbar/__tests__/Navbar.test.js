import React from "react";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Navbar from "./../Navbar";

it("Renders without Crashing", () => {
  render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );
});

it("Renders with correct text", () => {
  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );

  getByText("Home");
  getByText("Skills");
  getByText("Me");
  getByText("Contact");
});

it("Renders with correct button styling", () => {
  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );
  expect(getByText("Home")).toHaveClass("navbar-link");
  expect(getByText("Skills")).toHaveClass("navbar-link");
  expect(getByText("Me")).toHaveClass("navbar-link");
  expect(getByText("Contact")).toHaveClass("navbar-link");
});

it("Changes 'selected' className when clicking 'Skills' Button", () => {
  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );

  const SkillsButton = getByText("Skills");
  const HomeButton = getByText("Home");

  fireEvent.click(SkillsButton);

  expect(SkillsButton).toHaveClass("selected");
  expect(HomeButton).not.toHaveClass("selected");
});

it("Changes 'selected' className when clicking 'Me' Button", () => {
  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );

  const MeButton = getByText("Me");
  const HomeButton = getByText("Home");

  fireEvent.click(MeButton);

  expect(MeButton).toHaveClass("selected");
  expect(HomeButton).not.toHaveClass("selected");
});

it("Changes 'selected' className when clicking 'Contact' Button", () => {
  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );

  const ContactButton = getByText("Contact");
  const HomeButton = getByText("Home");

  fireEvent.click(ContactButton);

  expect(ContactButton).toHaveClass("selected");
  expect(HomeButton).not.toHaveClass("selected");
});

it("Sends to correct path once 'Home' button is clicked", () => {
  global.Window = { location: { pathname: null } };

  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );
  const HomeButton = getByText("Home");
  fireEvent.click(HomeButton);
  expect(global.window.location.pathname).toEqual("/Home");
});

it("Sends to correct path once 'Skills' button is clicked", () => {
  global.Window = { location: { pathname: null } };

  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );
  const SkillsButton = getByText("Skills");
  fireEvent.click(SkillsButton);
  expect(global.window.location.pathname).toEqual("/Skills");
});

it("Sends to correct path once 'Me' button is clicked", () => {
  global.Window = { location: { pathname: null } };

  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );
  const MeButton = getByText("Me");
  fireEvent.click(MeButton);
  expect(global.window.location.pathname).toEqual("/Me");
});

it("Sends to correct path once 'Contact' button is clicked", () => {
  global.Window = { location: { pathname: null } };

  const { getByText } = render(
    <Router>
      <Navbar></Navbar>
    </Router>
  );
  const ContactButton = getByText("Contact");
  fireEvent.click(ContactButton);
  expect(global.window.location.pathname).toEqual("/Contact");
});

it("Is correct compared to snapshot", () => {
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
