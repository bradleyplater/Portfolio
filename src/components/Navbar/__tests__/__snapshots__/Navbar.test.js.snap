// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is correct compared to snapshot 1`] = `
<nav
  className="navbar"
>
  <ul
    className="navbar-container"
  >
    <li
      className="navbar-list-item"
    >
      <a
        aria-current={null}
        className="navbar-link"
        data-testid="link.Home"
        href="/Home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navbar-list-item"
    >
      <a
        aria-current={null}
        className="navbar-link"
        data-testid="link.Skills"
        href="/Skills"
        onClick={[Function]}
      >
        Skills
      </a>
    </li>
    <li
      className="navbar-list-item"
    >
      <a
        aria-current={null}
        className="navbar-link"
        data-testid="link.Me"
        href="/Me"
        onClick={[Function]}
      >
        Me
      </a>
    </li>
    <li
      className="navbar-list-item"
    >
      <a
        aria-current="page"
        className="navbar-link navbar-link selected"
        data-testid="link.Contact"
        href="/Contact"
        onClick={[Function]}
        style={Object {}}
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
